src/routes/tutorial/[slug]/Output.svelte

```diff
@@ -0,0 +1,284 @@
+<script>
+	import { afterNavigate } from '$app/navigation';
+	import { onMount } from 'svelte';
+	import { browser, dev } from '$app/environment';
+	import Chrome from './Chrome.svelte';
+	import Loading from './Loading.svelte';
+	import { create_adapter } from './adapter';
+	import { state } from './state';
+
+	/** @type {string} */
+	export let path;
+
+	/** @type {HTMLIFrameElement} */
+	let iframe;
+	let loading = true;
+	let initial = true;
+
+	/** @type {Error | null} */
+	let error = null;
+
+	let progress = 0;
+	let status = 'initialising';
+
+	/** @type {import('$lib/types').Adapter} Will be defined after first afterNavigate */
+	let adapter;
+	/** @type {string[]} */
+	let history_bwd = [];
+	/** @type {string[]} */
+	let history_fwd = [];
+	let ignore_path_change = false;
+
+	function reset_history() {
+		history_bwd = [];
+		history_fwd = [];
+	}
+
+	onMount(() => {
+		const unsub = state.subscribe(async (state) => {
+			if (state.status === 'set' || state.status === 'switch') {
+				loading = true;
+
+				try {
+					clearTimeout(timeout);
+					await reset_adapter(state.stubs);
+
+					if (state.status === 'switch') {
+						set_iframe_src(adapter.base + path);
+					}
+
+					initial = false;
+				} catch (e) {
+					error = /** @type {Error} */ (e);
+					console.error(e);
+				}
+
+				loading = false;
+			} else if (state.status === 'update' && state.last_updated) {
+				const reload = await adapter.update([state.last_updated]);
+				if (reload) {
+					schedule_iframe_reload();
+				}
+			}
```

Comment:
this stuff feels racy — i feel like this logic belongs elsewhere, though i don't have a great design to offer right now

Reply:
The race conditions should be handled internally in `adapter.js`, though I now realize I'm not making use of the `"cancelled"` outcome here